### PR TITLE
Delete file from upload show

### PR DESCRIPTION
**Pull request by alfosco**
Adds working 'Delete File' button to upload show page.

### CODE REVIEW

File: app/views/uploads/show.html.erb [hunk 1]
```diff
@@ -8,7 +8,7 @@
   <hr>
   <p>file type:  <strong><%= @upload.content_type %></strong></p>
   <hr>
-  <p><%= link_to "Download", upload_download_path(@upload), class: "btn" %></p>
+  <p><%= link_to "Download", upload_download_path(@upload), class: "btn" %> <%= link_to "Delete File", upload_path(@upload), method: :delete, class: "btn"%></p>
```
Comment (cmacaulay) [inline app/views/uploads/show.html.erb]: 🥇 🎉 🌮 👏

File: spec/features/user/user_can_delete_a_file_they_own_spec.rb [hunk 2]
```diff
@@ -1,24 +1,43 @@
...
+    click_on "Elephant"
+
+    click_on "Delete File"
 
     expect(current_path).to eq("/home")
-    expect(page).to_not have_content("Flowers")
+    expect(page).to_not have_content("Elephant")
     expect(page).to have_content("File deleted.")
   end
 end
```
Comment (cmacaulay) [inline spec/features/user/user_can_delete_a_file_they_own_spec.rb]: 🎉 TDD

**Comment by cmacaulay** [comment]
@alfosco please fix merge conflicts and merge 🎉

**Comment by JonathanSR** [comment]
Nice implementation, it passes my review and will merge once merge conflict is fixed.

**Pull request merged**